packages/belos/src/BelosSolverFactory.hpp

```diff
@@ -74,7 +58,46 @@
 #include <stdexcept>
 #include <vector>
 
+// For convenience have a specialized form of registerSolverFactoryForLib()
+// called at construction of SolverFactoryParent which will pick up the proper
+// lib (belos, belosepetra, or belosepetra) based on the template parameters.
+// Then registerSolverFactoryForLib() will link to and register all the managers
+// for the proper lib. These includes were added for the specialization.
+
+// for belos lib
```

Comment:
I have some ideas about `Belos::MultiVec` and `Belos::Operator` that I'll leave for a separate issue / discussion.